File: src/main/java/com/pubnub/api/models/server/files/ListFilesResult.java
package com.pubnub.api.models.server.files;

import com.pubnub.api.models.consumer.files.PNUploadedFile;
import lombok.Data;
import lombok.NonNull;

import java.util.Collection;

@Data
public class ListFilesResult {
    @NonNull
    private final int count;
    private final String next;
    @NonNull
    private final int status;
    @NonNull
    private final Collection<PNUploadedFile> data;
}

File: src/main/kotlin/com/pubnub/api/models/server/files/ListFilesResult.kt
package com.pubnub.api.models.server.files

import com.pubnub.api.models.consumer.files.PNUploadedFile

data class ListFilesResult(
    val count: Int,
    val next: String?,
    val status: Int,
    val data: Collection<PNUploadedFile>
)
